catalog/spatial/geocoding/spatial-geocoding-feature/src/main/java/org/codice/ddf/spatial/geocoding/feature/CatalogFeatureIndexer.java

```diff
@@ -0,0 +1,179 @@
+import ddf.security.service.SecurityServiceException;
+import java.io.Serializable;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.codice.ddf.security.common.Security;
+import org.codice.ddf.spatial.geocoding.FeatureExtractionException;
+import org.codice.ddf.spatial.geocoding.FeatureExtractor;
+import org.codice.ddf.spatial.geocoding.FeatureIndexer;
+import org.codice.ddf.spatial.geocoding.FeatureIndexingException;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.filter.Filter;
+
+public class CatalogFeatureIndexer implements FeatureIndexer {
+
+  private Security security = Security.getInstance();
+
+  private CatalogFramework catalogFramework;
+
+  private FilterBuilder filterBuilder;
+
+  public void setSecurity(Security security) {
+    this.security = security;
+  }
+
+  public void setCatalogFramework(CatalogFramework catalogFramework) {
+    this.catalogFramework = catalogFramework;
+  }
+
+  public void setFilterBuilder(FilterBuilder filterBuilder) {
+    this.filterBuilder = filterBuilder;
+  }
+
+  @Override
+  public void updateIndex(
+      String resource, FeatureExtractor featureExtractor, boolean create, IndexCallback callback)
+      throws FeatureExtractionException, FeatureIndexingException {
+
+    if (create) {
+      removeExistingMetacards();
+    }
+
+    AtomicInteger count = new AtomicInteger(0);
+
+    final FeatureExtractor.ExtractionCallback extractionCallback =
+        (SimpleFeature feature) -> {
+          createOrUpdateMetacardForFeature(feature, create);
+          callback.indexed(count.incrementAndGet());
+        };
+
+    featureExtractor.pushFeaturesToExtractionCallback(resource, extractionCallback);
+  }
+
+  private void createOrUpdateMetacardForFeature(SimpleFeature feature, boolean create)
+      throws FeatureIndexingException {
+    try {
+      security.runWithSubjectOrElevate(
+          () -> {
+            Metacard metacard = null;
+
+            if (!create) {
+              metacard = findMetacardForFeature(feature);
+            }
+            if (metacard == null) {
+              metacard = createMetacardForFeature(feature);
+              catalogFramework.create(new CreateRequestImpl(metacard));
+            } else {
+              catalogFramework.update(new UpdateRequestImpl(metacard.getId(), metacard));
+            }
+            return null;
+          });
+    } catch (SecurityServiceException | InvocationTargetException e) {
+      throw new FeatureIndexingException(e.getMessage());
+    }
+  }
+
+  private Metacard createMetacardForFeature(SimpleFeature feature) throws FeatureIndexingException {
+    Metacard metacard = new MetacardImpl();
+
+    String countryCode = (String) feature.getAttribute("ISO_A3");
+    metacard.setAttribute(new AttributeImpl(Metacard.TITLE, countryCode));
+
+    WKTWriter writer = new WKTWriter();
```

Comment:
Might want to consider making this a global with `ThreadLocal`.  Otherwise we are making a new WKTWriter for each invocation of `createMetacardForFeature`

Reply:
Sounds good - didn't realize this was statistically significant compared to, for example, the catalog ops.